test_docker.sh

```diff
@@ -56,6 +57,12 @@ for DIR in containers/*; do
         TAG="hubmap/portal-container-$BASENAME:$VERSION"
         build_test $TAG $BASENAME
         if [ "$1" == 'push' ]; then
+          # If the version is not different, do not push a new version.
+          DIFFERENCE=`diff "$DIR/VERSION" <(curl https://raw.githubusercontent.com/hubmapconsortium/portal-containers/master/containers/$BASENAME/VERSION)`
+          if [ -z $DIFFERENCE ]; then
```

Comment:
```suggestion
          URL="https://raw.githubusercontent.com/hubmapconsortium/portal-containers/master/containers/$BASENAME/VERSION"
          if ! diff "$DIR/VERSION" <(curl "$URL"); then
```
I think you could just use the exit status from `diff`:
```
$ if ! diff <(echo A) <(echo A); then echo 'There is a difference'; fi
$ if ! diff <(echo A) <(echo B); then echo 'There is a difference'; fi
1c1
< A
---
> B
There is a difference
```
No-difference counts as success, ie 0 status, ie true.

Reply:
Ok, that sounds good.